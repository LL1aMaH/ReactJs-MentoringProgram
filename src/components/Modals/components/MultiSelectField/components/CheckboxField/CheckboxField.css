.container {
  display: inline-flex;
  position: relative;
  padding-left: 1.5rem;
  margin: 0.625rem 0.625rem 0.625rem 0;
  line-height: 1.25rem;
  -webkit-user-select: none;
  user-select: none;
}

.checkboxField {
  position: absolute;
  height: 1.125rem;
  width: 1.125rem;
  opacity: 0;
  left: 0;
  top: 50%;
  z-index: 1;
  transform: translateY(-50%);
}

.checkMark {
  position: absolute;
  top: 50%;
  left: 0;
  height: 1.125rem;
  width: 1.125rem;
  border: 1.5px solid var(--border-color);
  border-radius: 2px;
  transform: translateY(-50%);
}

.checkMark::after {
  content: '';
  position: absolute;
  display: none;
}

.container .checkMark::after {
  left: 4px;
  top: 1px;
  width: 6px;
  height: 8px;
  border: solid var(--text-secondary-color);
  border-width: 0 2px 2px 0;
  transform: rotate(45deg);
}

.checkboxField,
.label {
  cursor: pointer;
  outline: none;
  color: var(--secondary-dark-color);
}

.disabled {
  cursor: default;
  color: var(--disable--color);
}

.checkboxField:checked ~ .checkMark {
  border-color: var(--border-color);
  background-color: var(--secondary-dark-color);
}

.checkboxField:checked ~ .checkMark::after {
  display: block;
}

.checkboxField:disabled {
  cursor: default;

  &:checked ~ .checkMark {
    border-color: var(--disable-color);
    background-color: var(--disabled-color);
  }
}
