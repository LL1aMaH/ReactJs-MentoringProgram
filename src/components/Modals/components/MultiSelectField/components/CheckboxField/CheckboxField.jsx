import React, { memo } from 'react';

import classnames from 'classnames';

import styles from '../../../../../../../styles/CheckboxField.module.scss';

export const CheckboxField = memo(function CheckboxField(props) {
  const { id, label, onChange, value, checked = false, className, disabled = false } = props;
  const labelClassName = classnames(styles.label, {
    [styles.disabled]: disabled,
  });

  const containerClassName = `${styles.container} ${className}`;

  const handleCheckboxChange = () => {
    onChange(label);
  };

  return (
    <div className={containerClassName}>
      <input
        type="checkbox"
        id={id}
        onChange={handleCheckboxChange}
        value={value}
        checked={checked}
        className={styles.checkboxField}
        disabled={disabled}
        data-testid="checkbox"
      />
      <span className={styles.checkMark} />
      <label htmlFor={id} className={labelClassName}>
        {label}
      </label>
    </div>
  );
});
