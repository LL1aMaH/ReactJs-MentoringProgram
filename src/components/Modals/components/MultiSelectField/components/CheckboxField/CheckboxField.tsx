import React, { memo } from 'react';

import classnames from 'classnames';

import styles from './CheckboxField.css';

export type TCheckboxFieldProps = {
  id: string;
  label: string;
  onChange: (label: string) => void;
  value?: string | number;
  checked?: boolean;
  className?: string;
  disabled?: boolean;
};

export const CheckboxField = memo(function CheckboxField(props: TCheckboxFieldProps): JSX.Element {
  const { id, label, onChange, value, checked = false, className, disabled = false } = props;
  const labelClassName = classnames(styles.label, {
    [styles.disabled]: disabled,
  });

  const containerClassName = `${styles.container} ${className}`;

  const handleCheckboxChange = () => {
    onChange(label);
  };

  return (
    <div className={containerClassName}>
      <input
        type="checkbox"
        id={id}
        onChange={handleCheckboxChange}
        value={value}
        checked={checked}
        className={styles.checkboxField}
        disabled={disabled}
        data-testid="checkbox"
      />
      <span className={styles.checkMark} />
      <label htmlFor={id} className={labelClassName}>
        {label}
      </label>
    </div>
  );
});
