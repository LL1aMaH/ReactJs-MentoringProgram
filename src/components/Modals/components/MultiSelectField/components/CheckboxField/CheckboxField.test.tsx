import React from 'react';
import { render } from '@testing-library/react';

import { CheckboxField } from './CheckboxField';

describe('CheckboxField component', () => {
  const getComponent = ({ checked = false, disabled = false } = {}) =>
    render(
      <CheckboxField
        id="test"
        label="test"
        onChange={jest.fn()}
        checked={checked}
        disabled={disabled}
      />,
    );

  it('should render correctly checked component', () => {
    expect(getComponent({ checked: true }).asFragment()).toMatchSnapshot();
  });
  it('should render correctly disabled unchecked component', () => {
    expect(getComponent({ disabled: true }).asFragment()).toMatchSnapshot();
  });
});
