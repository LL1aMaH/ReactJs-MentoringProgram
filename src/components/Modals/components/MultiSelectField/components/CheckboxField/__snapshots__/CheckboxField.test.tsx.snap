// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField component should render correctly checked component 1`] = `
<DocumentFragment>
  <div
    class="container undefined"
  >
    <input
      checked=""
      class="checkboxField"
      data-testid="checkbox"
      id="test"
      type="checkbox"
      value=""
    />
    <span
      class="checkMark"
    />
    <label
      class="label"
      for="test"
    >
      test
    </label>
  </div>
</DocumentFragment>
`;

exports[`CheckboxField component should render correctly disabled unchecked component 1`] = `
<DocumentFragment>
  <div
    class="container undefined"
  >
    <input
      class="checkboxField"
      data-testid="checkbox"
      disabled=""
      id="test"
      type="checkbox"
      value=""
    />
    <span
      class="checkMark"
    />
    <label
      class="label disabled"
      for="test"
    >
      test
    </label>
  </div>
</DocumentFragment>
`;
